src/GraphQLParser/ParserContext.cs

```diff
@@ -898,6 +893,7 @@ private GraphQLTypeExtensionDefinition ParseTypeExtensionDefinition()
             return new GraphQLTypeExtensionDefinition
             {
                 Comment = comment,
+                Name = definition.Name,
```

Comment:
It was the only type inherited from `GraphQLTypeDefinition` that did not have a `Name`.